File: 11495.py
from sys import stdin

inversions = 0
aux = [None for _ in range(100005)]

def merge(nums, lo, hi):
    global inversionsCnt
    for x in range(lo, hi): aux[x] = nums[x]
    mid = (hi+lo)//2
    i, j, k = lo, mid, lo

    while(k < hi):
        if(i == mid):
            nums[k] = aux[j]
            j+=1
        elif(j == hi):
            nums[k] = aux[i]
            i+=1
        elif(aux[i] > aux[j]):
            inversionsCnt += mid-i
            nums[k] = aux[j]
            j+=1
        else:
            nums[k] = aux[i]
            i+=1
        k += 1

def mergeSort(nums, lo, hi):
    global inversionsCnt
    if(hi-lo > 1):
        mid = (hi+lo)//2

        mergeSort(nums, lo, mid)
        mergeSort(nums, mid, hi)
        merge(nums, lo, hi)



def main():
    global inversionsCnt

    nums = list(map(int, stdin.readline().split()))
    while(nums[0] != 0):
        

        inversionsCnt = 0
        mergeSort(nums, 1, len(nums))
        if(inversionsCnt%2==0):print("Carlos")
        else: print("Marcelo")
        nums = list(map(int, stdin.readline().split()))
main()

File: 11495.cpp
#include <bits/stdc++.h>
using namespace std;

int inversions;
vector<int> aux(100000, 0);


void merge(vector<int> &nums, int lo, int hi){
  for(int i = lo ; i < hi ; i++){
    aux[i] = nums[i];
  }
  int mid = (hi+lo)/2;
  int i = lo, j = mid, k = lo;

  while(k < hi){
    if(i == mid){
      nums[k] = aux[j];
      j+=1;
    }
    else if(j == hi){
      nums[k] = aux[i];
      i+=1;
    }
    else if(aux[i] > aux[j]){
      inversions += (mid - i);
      nums[k] = aux[j];
      j+=1;
    }
    else{
      nums[k] = aux[i];
      i+=1;
    }
    k += 1;
  }
}

void mergeSort(vector<int> &nums, int lo, int hi){
  if(hi-lo > 1){
    int mid = (hi+lo)/2;
    mergeSort(nums, lo, mid);
    mergeSort(nums, mid, hi);
    merge(nums, lo, hi);
  }
}



int main(){
  int n, num;
  while(true){
    cin >> n;
    if(n == 0) break;
    vector<int> nums;
    for(int i = 0 ; i < n ; i++){
      cin >> num;
      nums.push_back(num);
    }
    inversions = 0;
    mergeSort(nums, 0, nums.size());
    if(inversions%2==0) cout << "Carlos" << endl;
    else cout << "Marcelo" << endl;
  }
  return 0;
}